src/StaticReflection/SourceLocator/RenamedClassesSourceLocator.php

```diff
@@ -27,6 +27,12 @@ public function __construct(
 
     public function locateIdentifier(Reflector $reflector, Identifier $identifier): ?Reflection
     {
+        $identifierType = $identifier->getType();
+        $typeName = $identifierType->getName();
+        if ($typeName !== 'PHPStan\BetterReflection\Reflection\ReflectionClass') {
+            return null;
+        }
+
```

Comment:
It looks very weird. What is this related to?

Reply:
it is happen when renamed class is inside func call, eg:

```php
app(Queue::class);
```

without this, it will crash: 


```bash
There were 2 failures:

1) Rector\Tests\Renaming\Rector\Name\RenameClassRector\AutoImportNamesTest::test with data set #10 (Symplify\SmartFileSystem\SmartFileInfo Object (...))
assert($functionInfo instanceof \PHPStan\BetterReflection\Reflection\ReflectionFunction || $functionInfo === null) in phar:///Users/samsonasik/www/rector-src/vendor/phpstan/phpstan/phpstan.phar/vendor/ondrejmirtes/better-reflection/src/Reflector/DefaultReflector.php:59

2) Rector\Tests\Renaming\Rector\Name\RenameClassRector\AutoImportNamesTest::test with data set #2 (Symplify\SmartFileSystem\SmartFileInfo Object (...))
assert($functionInfo instanceof \PHPStan\BetterReflection\Reflection\ReflectionFunction || $functionInfo === null) in phar:///Users/samsonasik/www/rector-src/vendor/phpstan/phpstan/phpstan.phar/vendor/ondrejmirtes/better-reflection/src/Reflector/DefaultReflector.php:59

Reply:
We should figure out why this happens. This ignore seem to hard coded solution.

What about fixture with method call?

Reply:
in method call arg seems working ok https://github.com/rectorphp/rector-src/pull/2431/commits/525c3479acb3dadbf4d60a3317227d5eb22b2095 .

Reply:
The identifier type is `PHPStan\BetterReflection\Reflection\ReflectionFunction` if not class:

```bash
.PHPStan\BetterReflection\Identifier\IdentifierType #17599
   name: 'PHPStan\BetterReflection\Reflection\ReflectionFunction'
```

we can exclude it for `PHPStan\BetterReflection\Reflection\ReflectionFunction` only if we want to avoid the negation check:

```diff
-        if ($typeName !== 'PHPStan\BetterReflection\Reflection\ReflectionClass') {
+        if ($typeName === 'PHPStan\BetterReflection\Reflection\ReflectionFunction') {
```

so we can evaluate other type name if there is other use case like this again. what do you think?

Reply:
@TomasVotruba I found it 🎉 , it seems resolved by verify start line < 0: 

```diff
-        return ReflectionClass::createFromNode($classReflector, $class, $fakeLocatedSource);
+        $reflectionClass = ReflectionClass::createFromNode($classReflector, $class, $fakeLocatedSource);

+        if ($reflectionClass->getStartLine() < 0) {
+            return null;
+        }

+      return $reflectionClass;
```

so no need the hard code check https://github.com/rectorphp/rector-src/pull/2431/commits/b126df4cb80db4ee8dbf4e695b1eab7d535377c3

Reply:
It seems there is error that only in CI:

```diff
There was 1 failure:
1) Rector\Tests\Renaming\Rector\Name\RenameClassRector\RenameClassRectorTest::test with data set #13 (Symplify\SmartFileSystem\SmartFileInfo Object (...))
rules-tests/Renaming/Rector/Name/RenameClassRector/Fixture/rename_class.php.inc
Failed asserting that string matches format description.
--- Expected
+++ Actual
@@ @@
 class MyNewClass
 {
     /**
-     * @return \MyNewNamespace\MyNewClass
+     * @return MyClass
```

Reply:
I think the way to go is compare equal to `PHPStan\BetterReflection\Reflection\ReflectionFunction` so it should be safe as this class is checking renamed class, not renamed function https://github.com/rectorphp/rector-src/pull/2431/commits/4fe4bc79c1f51d8739f18728de31d7cccc7fb7b9

Reply:
@TomasVotruba it seems `RenamedClassesSourceLocator` can be completely removed since PHPStan 1.7.x is fully static reflection, let's remove it https://github.com/rectorphp/rector-src/pull/2431/commits/097753b1e742345bb3d3f460aa5719b2f4aa0a19

Reply:
Back to get error in CI : 

```diff
There was 1 failure:
1) Rector\Tests\Renaming\Rector\Name\RenameClassRector\RenameClassRectorTest::test with data set #13 (Symplify\SmartFileSystem\SmartFileInfo Object (...))
rules-tests/Renaming/Rector/Name/RenameClassRector/Fixture/rename_class.php.inc
Failed asserting that string matches format description.
--- Expected
+++ Actual
@@ @@
 class MyNewClass
 {
     /**
-     * @return \MyNewNamespace\MyNewClass
+     * @return MyClass
```

Reply:
I reverted the removal of `RenamedClassesSourceLocator`, back to use compare type name is `PHPStan\BetterReflection\Reflection\ReflectionFunction` https://github.com/rectorphp/rector-src/pull/2431/commits/4b1cbd4b2dd50dc1a215a107a5743f4f1ad70bb7

Reply:
@TomasVotruba ok, I found it 🎉 , it seems resolved by check : 

```php
        if (! $identifier->isClass()) {
            return null;
        }
```